ui/lib/pki/addon/components/page/pki-certificate-details.ts

```diff
@@ -0,0 +1,47 @@
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+import FlashMessageService from 'vault/services/flash-messages';
+import DownloadService from 'vault/services/download';
+import PkiCertificateBaseModel from 'vault/models/pki/certificate/base';
+
+interface Args {
+  model: PkiCertificateBaseModel;
+  onRevoke?: CallableFunction;
+  onBack?: CallableFunction;
+}
+
+export default class PkiCertificateDetailsComponent extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly download: DownloadService;
+
+  @action
+  downloadCert() {
+    try {
+      const formattedSerial = this.args.model.serialNumber?.replace(/(\s|:)+/g, '-');
+      this.download.pem(formattedSerial, this.args.model.certificate);
+      this.flashMessages.info('Your download has started.');
+    } catch (err) {
+      this.flashMessages.danger(errorMessage(err, 'Unable to prepare certificate for download.'));
+    }
+  }
+
+  @task
+  @waitFor
+  *revoke() {
+    try {
+      yield this.args.model.save();
```

Comment:
The comment in the adapter was helpful - it isn't super intuitive here that `.save()` equals `/cert/revoke`. It may be helpful to also add here that revoking is a certificates only POST operation or something. but total nit! 🧹

Reply:
Yes I agree. I considered adding a custom `revoke` method to the adapter but then you lose the automatic updating of the record that you get from `save` since `revocation_time` is returned from the request. I'll go ahead and add another comment here.